pages/[user]/[type].tsx

```diff
@@ -76,25 +81,33 @@ export default function Type(props): Type {
     setSelectedDate(dayjs().month(selectedMonth).date(day));
   };
 
+  const checkAvailableDays = (day) => {
+    if (selectedMonth > minMonth && selectedMonth < maxMonth) {
+      return true
+    }
+    else if (selectedMonth > minMonth && selectedMonth === maxMonth) {
+      return day <= maxDay
+    }
+    else if (selectedMonth === minMonth && selectedMonth < maxMonth) {
+      return day >= minDay
+    }
+    else if (selectedMonth === minMonth  && selectedMonth === maxMonth) {
+      return day >= minDay && day <= maxDay
+    }
+    else {
+      return false
+    }
+  }
+
   // Combine placeholder days with actual days
   const calendar = [
     ...emptyDays,
     ...days.map((day) => (
       <button
         key={day}
         onClick={() => changeDate(day)}
-        disabled={
-          selectedMonth < parseInt(dayjs().format("MM")) && dayjs().month(selectedMonth).format("D") > day
-        }
-        className={
-          "text-center w-10 h-10 rounded-full mx-auto " +
-          (dayjs().isSameOrBefore(dayjs().date(day).month(selectedMonth))
-            ? "bg-blue-50 text-blue-600 font-medium"
-            : "text-gray-400 font-light") +
-          (dayjs(selectedDate).month(selectedMonth).format("D") == day
-            ? " bg-blue-600 text-white-important"
-            : "")
-        }>
+        disabled={selectedMonth < parseInt(dayjs().format('MM')) && dayjs().month(selectedMonth).format('D') > day || minDay !== null ? !checkAvailableDays(day) : '' } 
```

Comment:
Codacy has a fix for the issue: [Replace `selectedMonth·<·parseInt(dayjs().format('MM'))·&&·dayjs().month(selectedMonth).format('D')·>·day·||·minDay·!==·null·?·!checkAvailableDays(day)·:·''·}·` with `⏎··········(selectedMonth·<·parseInt(dayjs().format("MM"))·&&⏎············dayjs().month(selectedMonth).format("D")·>·day)·||⏎··········minDay·!==·null⏎············?·!checkAvailableDays(day)⏎············:·""⏎········}`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
        disabled={
```